extension/content/components/recipes/form/arguments/BranchedAddon.js

```diff
@@ -0,0 +1,218 @@
+          <ToggleField label="Prevent New Enrollment" name="isEnrollmentPaused">
+            Prevents new users from joining this study cohort. Exisiting users
+            will remain in the study.
+          </ToggleField>
+        </Col>
+      </Row>
+      <Branches options={options} />
+    </FormGroup>
+  );
+}
+
+function Branches({ options }) {
+  const data = useRecipeDetailsData();
+  const dispatch = useRecipeDetailsDispatch();
+
+  const handleClickAddBranch = () => {
+    /** @type {{ratio: number, slug: string, value: string | number | boolean}} */
+    const newBranch = { ratio: 1, slug: "", extensionApiId: "" };
+
+    dispatch({
+      type: ACTION_UPDATE_DATA,
+      data: {
+        ...data,
+        arguments: {
+          ...data.arguments,
+          branches: [...data.arguments.branches, newBranch],
+        },
+      },
+    });
+  };
+
+  let branchesList = <HelpBlock>There are no branches.</HelpBlock>;
+  if (data.arguments.branches && data.arguments.branches.length) {
+    branchesList = data.arguments.branches.map((branch, index) => (
+      <Branch key={index} index={index} options={options} />
+    ));
+  }
+
+  return (
+    <FormGroup>
+      <ControlLabel>Branches</ControlLabel>
+      <Panel bordered>
+        {branchesList}
+        <Divider />
+        <IconButton
+          icon={<Icon icon="plus-circle" />}
+          onClick={handleClickAddBranch}
+        >
+          Add Branch
+        </IconButton>
+      </Panel>
+    </FormGroup>
+  );
+}
+
+function Branch({ index, options }) {
+  const data = useRecipeDetailsData();
+  const dispatch = useRecipeDetailsDispatch();
+  const branch = data.arguments.branches[index];
+  const { branches } = data.arguments;
+
+  const handleClickDelete = () => {
+    dispatch({
+      type: ACTION_UPDATE_DATA,
+      data: {
+        ...data,
+        arguments: {
+          ...data.arguments,
+          branches: branches.filter((b, i) => i !== index),
+        },
+      },
+    });
+  };
+
+  const handleChange = (name, cb = (v) => v) => {
+    return (value) => {
+      dispatch({
+        type: ACTION_UPDATE_DATA,
+        data: {
+          ...data,
+          arguments: {
+            ...data.arguments,
+            branches: branches.map((b, i) => {
+              if (i === index) {
+                return { ...b, [name]: cb(value) };
+              }
+
+              return b;
+            }),
+          },
+        },
+      });
+    };
+  };
+
+  const parseNumericInput = (value) => {
+    if (!value) {
+      return "";
+    }
+
+    return parseInt(value, 10);
+  };
+
+  return (
+    <FormGroup>
+      <div className="d-flex">
+        <div className="pr-2">
+          <ControlLabel>&nbsp;</ControlLabel>
+          <IconButton
+            circle
+            color="red"
+            icon={<Icon icon="trash" />}
+            size="sm"
+            onClick={handleClickDelete}
+          />
+        </div>
+        <div className="pr-1">
```

Comment:
add `flex-grow-1` here

Reply:
I think it gets a little weird with flex grow?
![Screen Shot 2020-04-29 at 9 20 49 PM](https://user-images.githubusercontent.com/25379936/80671978-35a94280-8a60-11ea-801d-b89a959b4680.png)

Reply:
ah right,  i would set these both to `flex-basis-0 flex-grow-1` and that should fix it. however `flex-basis-0` only exists on my local branch for the recipe details page so it wouldn't work till that lands but that's okay.